File: MappingDemo/src/main/java/com/phunware/mapping/sample/ui/MainActivity.java
package com.phunware.mapping.sample.ui;

import android.os.Bundle;
import android.support.v7.app.ActionBarActivity;
import android.view.Window;

import com.phunware.mapping.sample.R;

public class MainActivity extends ActionBarActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        supportRequestWindowFeature(Window.FEATURE_INDETERMINATE_PROGRESS);
        setContentView(R.layout.activity_main);
    }
}

File: Samples/kotlinsample/src/main/java/com/phunware/kotlinsample/MainActivity.kt
package com.phunware.kotlinsample

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView

class MainActivity : AppCompatActivity(), DemoAdapter.DemoOnClickListener {
    private var demoAdapter: DemoAdapter? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.main)

        // Instantiate all demo info
        val demoDetails = DemoDetailsList(this)

        // Create list of all demos
        val demoRecyclerView = findViewById<RecyclerView>(R.id.demo_list)
        val layoutManager = LinearLayoutManager(this)
        demoRecyclerView.layoutManager = layoutManager
        demoAdapter = DemoAdapter(demoDetails.getDemos(), this)
        demoRecyclerView.adapter = demoAdapter
    }

    override fun onItemClicked(title: String) {
        val demo = demoAdapter!!.getItem(title)
        if (demo != null) {
            startActivity(Intent(this, demo.activityClass))
        }
    }
}
